src/org/labkey/test/components/ui/search/ManageSampleFinderViewsModal.java

```diff
@@ -0,0 +1,158 @@
+
+    protected ManageSampleFinderViewsModal(String title, WebDriver driver)
+    {
+        super(new ModalDialog.ModalDialogFinder(driver).withTitle(title));
+    }
+
+    public void editViewName(String viewName)
+    {
+        WebElement editIcon = getViewEditIcon(viewName);
+        if (editIcon != null)
+        {
+            editIcon.click();
+        }
+        clearElementCache();
+        elementCache(); // waitForReady
+    }
+
+    public void deleteView(String viewName)
+    {
+        WebElement deleteIcon = getViewDeleteIcon(viewName);
+        if (deleteIcon != null)
+        {
+            deleteIcon.click();
+        }
+        clearElementCache();
+        elementCache(); // waitForReady
+    }
+
+    public void deleteAllViews()
+    {
+        List<String> views = getViews();
+        Collections.reverse(views);
+        for (String view : views)
+        {
+            deleteView(view);
+        }
+    }
+
+    public List<String> getViews()
+    {
+        return Locator.tagWithClass("div", "small-margin-bottom")
+                .findElements(getComponentElement())
+                .stream()
+                .map(WebElement::getText)
+                .collect(Collectors.toList());
+    }
+
+    public WebElement getView(String viewName)
+    {
+        List<WebElement> webElements = Locator.tagWithClass("div", "small-margin-bottom")
+                .findElements(getComponentElement());
+
+        for (WebElement element : webElements)
+        {
+            if (element.getText().equals(viewName))
+                return element;
+        }
+
+        return null;
+    }
+
+    public WebElement getViewIcon(String reportName, String iconCls)
+    {
+        WebElement viewRow = getView(reportName);
+        if (viewRow == null)
+            return null;
+
+        return Locator
+                .tagWithClass("i", iconCls)
+                .findElementOrNull(viewRow);
+    }
+
+    public WebElement getViewEditIcon(String reportName)
+    {
+        return getViewIcon(reportName, "fa-pencil");
+    }
+
+    public WebElement getViewDeleteIcon(String reportName)
+    {
+        return getViewIcon(reportName, "fa-trash-o");
+    }
+
+    public boolean isViewLocked(String reportName)
+    {
+        return getViewLockIcon(reportName) != null && getViewEditIcon(reportName) == null;
+    }
+
+    public WebElement getViewLockIcon(String reportName)
+    {
+        return getViewIcon(reportName, "fa-lock");
+    }
+
+    public String getInputValue()
+    {
+        return Locator.tag("input").findElement(getComponentElement()).getAttribute("value");
+    }
+
+    public ManageSampleFinderViewsModal setName(String name)
+    {
+        WebElement input = elementCache().nameInput;
+        getWrapper().actionClear(input);
+        input.sendKeys(name);
+        input.sendKeys(Keys.TAB);
+        return this;
+    }
+
+    public void clickDone()
+    {
+        elementCache().doneBtn.click();
+        waitForClose();
+    }
```

Comment:
Because this extends ModalDialog you don't need to implement this method. The test can simple call the dismiss method with the button text. Alternatively you could change this method to:
```
    public void clickDone()
    {
        dismiss("Done editing");
    }

```
Calling the dismiss method will also call the waitForClose method.